### ISSUE TITLE
Unit Testing

### ISSUE THREAD

**Issue by gblanchard4** [benjjneb/dada2#187]
Would you like to start trying to integrate some unit testing into the package? I've been looking at [framework suggestions](http://bioconductor.org/developers/unitTesting-guidelines/) BioConductor's site. It looks like RUnit and Testthat are the two options. Phyloseq seems to be using the Testthat framework.

Although labor intensive, I think they could be beneficial, especially with all the new features being added.

**Comment by benjjneb** [comment]
Yes this would definitely be a good thing and really is overdue. I think testthat is the way to go.

The question I have right now is that some testing should use moderate size datasets (eg. a mock community fastq) which are too large to include in the package...

**Comment by gblanchard4** [comment]
Hmmm good point. The MiSeq SOP is ~37MB. GitHubs (working with large files documentation)[https://help.github.com/articles/working-with-large-files/] states 
> GitHub will warn you when pushing files larger than 50 MB. You will not be allowed to push files larger than 100 MB.

So putting the MiSeqSOP tarball would fit, but it may slow everything down. I will have to test that. Potentially you could pull the tarball down using curl/wget as part of setup for the tests.

**Comment by gblanchard4** [comment]
Thinking about this more, a decent amount of the the methods could be tested with just the fastqs included in extdata. At least enough to start getting code coverage up. I am playing around with testthat now to see where I start hitting snags.

**Comment by benjjneb** [comment]
More complete unit-testing with testthat is still desirable, but won't make it into 1.6.

I have expanded the running examples in several functions (i.e. `assignTaxonomy` and `assignSpecies`) which effectively serve during R CMD CHECK to ensure that functions at least run w/o error with expected arguments.
